### ISSUE TITLE
Allow configuring http

### ISSUE THREAD

**Issue by ben-xD** [libsql/libsql-client-ts#39]
When using a local database, e.g. `libsql://localhost:3030`, this library would use `https`. This fails because I can't access localhost over https. Please allow me to disable https, so that I can connect to a local client over HTTP. I'm using Cloudflare Workers so it seems like I cannot use the file directly.

## Error
```
[0] workerd/jsg/util.c++:276: error: e = kj/compat/tls.c++:74: failed: OpenSSL error; message = error:100000f7:SSL routines:OPENSSL_internal:WRONG_VERSION_NUMBER
[0] stack: 1027b92ab 1027bbb1f 1027bc23b 1027bc968 1027c47f8 103760d3c 103762998 1028b3304 1028b3728 1028b2eec 1037506d0 1024495d0 1024468d4 10251c2fc 10245df10 1
```

The error seems very similar to https://stackoverflow.com/a/63321432/7365866

There some instructions on adding https for localhost https://web.dev/how-to-use-local-https/, but that requires `turso dev` to use the certificate.

**Comment by honzasp** [comment]
The `libsql:` scheme always implies TLS (either `https:` or `wss:`). If you want to use a plaintext protocol, then you can pass a `http:` or `ws:` URL to the client (so "http://localhost:3030" or "ws://localhost:3030" in your case). Unfortunately, there is no plaintext variant of `libsql:` that would allow the library to choose between `http:` and `ws:` automatically. We discussed this problem just yesterday, but we have yet to find a satisfactory solution.

**Comment by ben-xD** [comment]
Oh, I have tried `http:` in the past (mentioned in https://github.com/chiselstrike/turso-cli/issues/418#issuecomment-1555813513) but unfortunately I just a get a 404 error (the page returns 404 anyway if you visit `http://locahost:3030` in the browser).

I'm a bit stuck on running a local database for testing.

**Comment by honzasp** [comment]
Ah, so you are using `turso dev`? That supports only a very outdated version of the HTTP API, the libsql client won't be able to talk to it. The only viable option for local development right now is to run `sqld`, which is the open-source part of Turso:
https://github.com/libsql/sqld

**Comment by ben-xD** [comment]
Oh I didn't know that. I'll try that, thanks

**Comment by honzasp** [comment]
We decided to add a `?tls=0` query parameter to disable TLS with `libsql:` URLs (so `http://localhost:3030?tls=0` will connect with either HTTP or WebSockets to localhost on port 3030 without TLS). I hope to implement this soon, but in the meantime, you can use `http:` or `ws:` explicitly.

**Comment by ben-xD** [comment]
I've tried sqld, connecting using `ws://127.0.0.1:8080`, but I get `SQL_MANY_STATEMENTS`.

1. I have only 1 migration file, and it has 3 SQL commands. This file is generated by drizzle-kit. Unfortunately, I get an error:
```
file:///Users/zen/repos/batmark/node_modules/.pnpm/@libsql+client@0.1.6/node_modules/@libsql/client/lib-esm/hrana.js:331
        return new LibsqlError(e.message, code, e);
               ^
LibsqlError: SQL_MANY_STATEMENTS: SQL string contains more than one statement
    at mapHranaError (file:///Users/zen/repos/batmark/node_modules/.pnpm/@libsql+client@0.1.6/node_modules/@libsql/client/lib-esm/hrana.js:331:16)
    at HranaTransaction.execute (file:///Users/zen/repos/batmark/node_modules/.pnpm/@libsql+client@0.1.6/node_modules/@libsql/client/lib-esm/hrana.js:243:19)
    at processTicksAndRejections (node:internal/process/task_queues:95:5) {
  code: 'SQL_MANY_STATEMENTS',
  [cause]: [ResponseError: SQL string contains more than one statement] {
    code: 'SQL_MANY_STATEMENTS',
    proto: {
      message: 'SQL string contains more than one statement',
      code: 'SQL_MANY_STATEMENTS'
    }
  }
}
```

**Comment by CodingDoug** [comment]
@ben-xD  Since you have a new issue that's unrelated to the one you originally posted, could you do us a huge favor and post that as a new issue? That will make it easier for us to track.

**Comment by ben-xD** [comment]
Sorry, I could've created a new issue for the `SQL_MANY_STATEMENTS` error, but I am not using Turso right now. I wouldn't want to create an issue that is outdated or not useful to any user. I just noticed one has been created: https://github.com/libsql/libsql-client-ts/issues/43 🙏

I switched to Postgres/Neon because of some rough edges with SQLite/turso, especially wrt. running things locally (which is important since I'm travelling). I can run Postgres locally without neon.

**Comment by honzasp** [comment]
Hi Ben, I am sorry about that! We are working on improving the `turso dev` experience, so that it spawns a full-fledged sqld instance that is fully compatible with Turso. In the meantime, you can get full local experience by running sqld directly.

**Comment by CodingDoug** [comment]
@ben-xD please don't worry about filing issues that you feel are not helpful for other users.  The issues are helpful for us to track how people are using the product, and the specific cases that are problematic, even if solutions appear to be in progress. It's never a bad idea from our perspective to detail the specific steps you're taking to reproduce some unexpected behavior!

There are a number of ways to run sqld locally now (including an improved `turso dev`).  If you have time, your feedback would be helpful on the supported options so far: https://github.com/libsql/sqld/blob/main/docs/BUILD-RUN.md

**Comment by honzasp** [comment]
Fixed in 0.2.0
